config/initializers/replace_references_service.rb

```diff
@@ -37,6 +37,7 @@
       "CostEntry" => %i[logged_by_id user_id],
       "CostQuery" => :user_id,
       "::Doorkeeper::Application" => :owner_id,
+      "Document" => %i[author_id assigned_to_id responsible_id],
```

Comment:
If you have the patience to draw how this service is used I'm curious.

_I might have a look on my own later, though_

Reply:
Sure, this service replaces deleted user references with a "delete user" reference- mitigates user deletion foreign key errors. https://github.com/opf/openproject/blob/03e187907db356c0173246a7cdfe8bf75d3183de/app/workers/principals/delete_job.rb#L48-L53